### DISCUSSION TITLE
Filter table based on Relationship Filter

### DISCUSSION THREAD

**Discussion by magickaito** [Budibase/budibase#5955] [Help] [answered]
Hi guys,

I am learning this fantastic framework. 

I am trying to reproduce my problem in a simple scenario like this:
- There is a Client Table. This table has one column "Company Name"
- There is an Invoice Table. This table has one column "Amount", and a relationship column "Client" that records which client this invoice belongs to. 

I auto-generated the CRUD screens, so far so good. 

Then in the invoices listing screen, I need a dropdown filter that let users **filter invoices by clients.** 

In the /invoice listing screen, I created a Form (Invoice schema) -> Relationship Picker (Points to Client field in the Invoice schema). And this works, I can see the dropdown listing all the possible clients:

![Screenshot 2022-05-18 at 7 25 14 PM](https://user-images.githubusercontent.com/1195573/169028113-530a5c5a-ce89-430a-8c4f-4c38c7cab114.jpg)

Next I need to actually filter the table based on the selected client.

When I select the Data Provider of the invoice table, click on Define Filter, this is what I see:

![Screenshot 2022-05-18 at 7 26 16 PM](https://user-images.githubusercontent.com/1195573/169028329-0550a871-7504-4b29-8190-a9190136f4ee.jpg)

I am expecting this UI to let me select filter by Client, but there is no Client options there. 

Why?

You can see the demo here: https://tplus.budibase.app/app/electrical-inspection#/invoice

**Comment by melohagan** [comment, marked as answer]
Hi @magickaito 

Thank you for the nice words, but unfortunately this isn't possible: https://github.com/Budibase/budibase/issues/5585

You may be able to use the JSON Column https://docs.budibase.com/docs/json as a workaround, otherwise we generally recommend using an RDBMS such as MySQL or PostgreSQL if you want to make full use of data relationships.

**Reply by magickaito** [reply]
Thanks for pointing me to the right direction.
